h2/connection.py

```diff
@@ -512,6 +512,7 @@ def _begin_new_stream(self, stream_id, allowed_ids):
             self.remote_settings.initial_window_size
         )
         s.inbound_flow_control_window = self.local_settings.initial_window_size
+        s.config = self.config
```

Comment:
This looks like it just gets casually patched onto the `H2Stream`. Can we update `H2Stream.__init__` to create this member variable initialized to `None`?